lambdas/pkgpush/index.py

```diff
@@ -360,3 +418,127 @@ def get_pkg(src_registry, data):
         pkg_max_size=PKG_FROM_FOLDER_MAX_PKG_SIZE,
         pkg_max_files=PKG_FROM_FOLDER_MAX_FILES,
     )
+
+
+def large_request_handler(request_type):
+    user_request_key = f'user-requests/{request_type}'
```

Comment:
I don't understand how this is thread safe at all. What happens if multiple users issue concurrent requests of the same type? Won't the be reading/deleting the same file in the TemporaryFile loop below?

Reply:
Users use the same key to write requests, but they write to the versioned bucket created by us via Cloudformation, so each request get its own ID (version ID of object). A single lambda instance process only a single request at the same time.